JetStreamCompose/jetstream/src/main/java/com/google/jetstream/presentation/screens/movies/MovieDetailsScreen.kt

```diff
@@ -52,26 +54,48 @@ object MovieDetailsScreen {
 
 @Composable
 fun MovieDetailsScreen(
-    movieId: String,
     goToMoviePlayer: () -> Unit,
     onBackPressed: () -> Unit,
     refreshScreenWithNewMovie: (Movie) -> Unit
 ) {
-    val childPadding = rememberChildPadding()
-    val screenHeight = LocalConfiguration.current.screenHeightDp.dp
-    val movieRepository = LocalMovieRepository.current!!
-    val movieDetails = remember { movieRepository.getMovieDetails(movieId = movieId) }
+    val movieDetailsScreenViewModel: MovieDetailsScreenViewModel =
+        viewModel(factory = MovieDetailsScreenViewModel.Factory)
+
+    val loadingState by movieDetailsScreenViewModel.loadingState.collectAsState()
+
+    when (loadingState) {
+        is LoadingState.Loading -> {
+            Text(text = "Loading...")
+        }
 
-    LaunchedEffect(Unit) {
-        movieRepository.getMovieDetails(movieId = movieId)
+        is LoadingState.Done -> {
+            Details(
+                movieDetails = (loadingState as LoadingState.Done).movieDetails,
+                goToMoviePlayer = goToMoviePlayer,
+                onBackPressed = onBackPressed,
+                refreshScreenWithNewMovie = refreshScreenWithNewMovie,
+                modifier = Modifier
+                    .fillMaxSize()
+                    .animateContentSize()
+            )
+        }
     }
+}
 
-    BackHandler(onBack = onBackPressed)
+@Composable
+private fun Details(
+    movieDetails: MovieDetails,
+    goToMoviePlayer: () -> Unit,
+    onBackPressed: () -> Unit,
+    refreshScreenWithNewMovie: (Movie) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    val childPadding = rememberChildPadding()
+    val screenHeight = LocalConfiguration.current.screenHeightDp.dp
```

Comment:
What is `screenHeight` needed for?

Reply:
The value is used to add some space to the bottom of the screen:
https://github.com/android/tv-samples/blob/main/JetStreamCompose/jetstream/src/main/java/com/google/jetstream/presentation/screens/movies/MovieDetailsScreen.kt#L153